### PR TITLE
[Feature] AppCenter Crash Logging

### PR DESCRIPTION
**Pull request by palatter**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[X ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by vhuseinova-msft — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Info.plist [hunk 1]
```diff
@@ -40,6 +40,8 @@
 	<array>
 		<string>audio</string>
 	</array>
+	<key>appCenterSecret</key>
+	<string>${appCenterSecret}</string>
```
Comment (kevinyulu) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Info.plist:44]: Can this secret value be a fixed one in here? This secret value won't change for debug builds from App Center

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Info.plist:44]: As a best practice we should not expose any secrets

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift [hunk 2]
```diff
@@ -7,17 +7,16 @@ import UIKit
...
-    let envConfigSubject = EnvConfigSubject()
 
     func scene(_ scene: UIScene,
                willConnectTo session: UISceneSession,
                options connectionOptions: UIScene.ConnectionOptions) {
         guard let windowScene = (scene as? UIWindowScene) else {
             return
         }
-        let entryViewController = EntryViewController(envConfigSubject: envConfigSubject)
+        let appDelegate = UIApplication.shared.delegate as? AppDelegate
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift:18]: suggestion ```
        guard let windowScene = (scene as? UIWindowScene),
              let appDelegate = UIApplication.shared.delegate as? AppDelegateelse {
            return
        }
        let entryViewController = EntryViewController(envConfigSubject: appDelegate.envConfigSubject)
```

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift:18]: To remove the force unwrap

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift [hunk 3]
```diff
@@ -27,16 +26,18 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
         }
 
     }
 
     func scene(_ scene: UIScene,
                openURLContexts URLContexts: Set<UIOpenURLContext>) {
 
+        let appDelegate = UIApplication.shared.delegate as? AppDelegate
+
         if let queryDict = URLContexts.first?.url.toQueryDictionary() {
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift:39]:
```suggestion
        if let queryDict = URLContexts.first?.url.toQueryDictionary(), 
        let appDelegate = UIApplication.shared.delegate as? AppDelegate {
```

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift:39]: So we can remove the force unwrap

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Info.plist (hunk 1 above)
Comment (palatter) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Info.plist:44]: Ya we can't have any secrets checked into the repo, which is why I injected it with pipeline.

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift (hunk 2 above)
Comment (palatter) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/SceneDelegate.swift:18]: Good call

**Review by kevinyulu — APPROVED** [review]

**Review by JoshuaLai — APPROVED** [review]
looks good

**Pull request merged**